### ISSUE TITLE
'-' and '_' in nested parsers

### ISSUE THREAD

**Issue by stasbel** [omni-us/jsonargparse#42]
This does work:

```python3
from jsonargparse import ArgumentParser, ActionParser
p1 = ArgumentParser()
p1.add_argument('--op1_like')

p2 = ArgumentParser()
p2.add_argument('--op2_like', action=ActionParser(parser=p1))

p3 = ArgumentParser()
p3.add_argument('--op3', action=ActionParser(parser=p2))

>>> p3.parse_args(['--op3.op2_like.op1_like=1'])
```

But this doesn't work:

```python3
from jsonargparse import ArgumentParser, ActionParser
p1 = ArgumentParser()
p1.add_argument('--op1-like')

p2 = ArgumentParser()
p2.add_argument('--op2-like', action=ActionParser(parser=p1))

p3 = ArgumentParser()
p3.add_argument('--op3', action=ActionParser(parser=p2))

>>> p3.parse_args(['--op3.op2-like.op1-like=1'])
# error: Unrecognized arguments: --op3.op2-like.op1-like=1
```

**Comment by mauvilsa** [comment]
Thank you for reporting! I have analyzed the issue and now know what the problem is and maybe a possible solution. Though right now I don't have the time to fix it. Will take a look at it again tomorrow.

**Comment by mauvilsa** [comment]
After analyzing this a bit more I have noticed that the way ActionParser works internally can be simplified greatly, making it easy to fix this issue and https://github.com/omni-us/jsonargparse/issues/41. Though it will take me a few days to have this ready.

**Comment by mauvilsa** [comment]
It wasn't as time consuming as I thought. This is now fixed and part of release v3.7.0.
